superset_wfs_dialect/base.py

```diff
@@ -667,19 +754,50 @@ def fetchone(self):
 
     def fetchmany(self, size=1):
         end = self._index + size
-        rows = [self._get_row_values(row) for row in self.data[self._index:end]]
+        rows = [self._get_row_values(row) for row in self.data[self._index : end]]
         self._index = min(end, len(self.data))
         return rows
 
     def close(self):
         pass
 
+    def _convert_value(self, value: str, type_name: str) -> Any:
+        """
+        Converts a string value to the corresponding Python type based on the XSD type
+        """
+        if value is None:
+            return None
+
+        try:
+            type_conversions = {
+                "string": str,
+                "int": int,
+                "integer": int,
+                "short": int,
+                "byte": int,
+                "float": float,
+                "double": float,
+                "decimal": float,
+                "long": float,
+                "boolean": lambda x: x.lower() in ("true", "1", "t", "y", "yes"),
+                "date": lambda x: datetime.strptime(x, "%Y-%m-%d").date(),
+                "dateTime": lambda x: datetime.strptime(x, "%Y-%m-%dT%H:%M:%S"),
```

Comment:
Did you check if casting of the date types works?

Reply:
[38f4032](https://github.com/terrestris/superset_wfs_dialect/pull/41/commits/38f4032aaef35ed9d6aa293626db56fc24c47936)